Sources/PackageGraph/PackageGraphLoader.swift

```diff
@@ -116,13 +117,33 @@ public struct PackageGraphLoader {
         // Connect up cross-package module dependencies.
         fillModuleGraph(packages)
     
-        let rootPackage = packages.last!
-        let externalPackages = packages.dropLast(1)
+        let (rootPackages, externalPackages) = packages.split { rootManifests.contains($0.manifest) }
 
         let modules = try recursiveDependencies(packages.flatMap{ map[$0] ?? [] })
         let externalModules = try recursiveDependencies(externalPackages.flatMap{ map[$0] ?? [] })
 
-        return PackageGraph(rootPackage: rootPackage, modules: modules, externalModules: Set(externalModules))
+        return PackageGraph(rootPackages: rootPackages, modules: modules, externalModules: Set(externalModules))
+    }
+}
+
+// FIXME: Maybe lift this into Utility.
+private extension Array {
```

Comment:
Yes, definitely, this should go into a more generic module.  Looks useful, though.